configure

```diff
@@ -37726,8 +37724,9 @@ CXXFLAGS=`echo $WXCONFIG_CXXFLAGS $C_AND_CXX_FLAGS $CXXFLAGS `
 OBJCFLAGS=`echo $WXCONFIG_CFLAGS $CWARNINGS $C_AND_CXX_FLAGS $OBJCFLAGS `
 OBJCXXFLAGS=`echo $WXCONFIG_CXXFLAGS $C_AND_CXX_FLAGS $OBJCXXFLAGS `
 
-WXCONFIG_CPPFLAGS=`echo $WXCONFIG_CPPFLAGS $WXCONFIG_ONLY_CPPFLAGS`
-
+if test "SHARED" = 1; then
+    WXCONFIG_CPPFLAGS="$WXCONFIG_CPPFLAGS -DWXUSINGDLL"
+fi
```

Comment:
I notice this no longer uses the `echo $X` "pattern", which appears to be done to "remove the extra white space" according to a comment elsewhere.  However, you can't safely just `echo $X` since echo itself takes command line arguments on some platforms - e.g. on linux with bash:

```
$ WXCONFIG_CFLAGS='-e -f -g'
$ WXCONFIG_CFLAGS=`echo $WXCONFIG_CFLAGS`
$ export WXCONFIG_CFLAGS
$ printenv WXCONFIG_CFLAGS
-f 
```

There are also issues with some shells if the value contains backslashes.

This change actually fixes such bugs for this case, but more generally it's probably better not to do this at all - extra whitespace is just cosmetic, but losing options like above could be really problematic.  But if the whitespace cleaning is felt to be desirable, something like `printf '%s' $WXCONFIG_CFLAGS` would probably be safe (the autoconf manual suggests that's portable).

Reply:
Thanks, this is a good point, and if autoconf thinks `printf` is portable, we should definitely use it.  I'll do it in another commit.